src/dataflow/src/sink/kafka.rs

```diff
@@ -618,6 +618,14 @@ where
             return true;
         }
 
+        if !pending_rows.is_empty() {
+            // We have some more rows that we need to wait for frontiers to advance before we
+            // can write them out. Let's make sure to reschedule with a small delay to give the
+            // system time to advance.
+            s.activator.activate_after(Duration::from_millis(100));
```

Comment:
I would assume `s.activator.activate()` is to aggressive and wouldn't give the rest of the system enough time? Although with doing that we would get re-activated on the next swing through the operators, which might be enough for the system to record advancement in durability information.

((I don't know enough about timely operator scheduling, so ... 🤷‍♂️ ))